editor.planx.uk/src/ui/shared/Checkbox.tsx

```diff
@@ -4,54 +4,60 @@ import React from "react";
 import { borderedFocusStyle } from "theme";
 
 const Root = styled(Box, {
-  shouldForwardProp: (prop) => !["disabled"].includes(prop.toString()),
-})<BoxProps & { disabled?: boolean }>(({ theme, disabled }) => ({
-  display: "inline-flex",
-  flexShrink: 0,
-  position: "relative",
-  width: 40,
-  height: 40,
-  borderColor: theme.palette.text.primary,
-  border: "2px solid",
-  backgroundColor: theme.palette.common.white,
-  "&:focus-within": borderedFocusStyle,
-  ...(disabled && {
-    border: `2px solid ${theme.palette.grey[400]}`,
-    backgroundColor: theme.palette.grey[400],
+  shouldForwardProp: (prop) =>
+    !["disabled", "variant"].includes(prop.toString()),
+})<BoxProps & { disabled?: boolean; variant?: "default" | "compact" }>(
+  ({ theme, disabled, variant }) => ({
+    display: "inline-flex",
+    flexShrink: 0,
+    position: "relative",
+    width: variant === "compact" ? 24 : 40,
+    height: variant === "compact" ? 24 : 40,
```

Comment:
Thanks for the added guidance @DafyddLlyr , makes a lot more sense to approach grouping of styles this way.

I've updated the property to be an optional `variable` prop, however after several attempts I have not been able to move the styling to `theme.ts` and use a native variant in this way (similar to how we use radio buttons). This is due to our creation of a custom checkbox component, in order to use our custom styling and tick icons, rather than the default provided by MUI.

Would you consider this worth merging in for now? With the view to being picked up as a further piece of work to move to using MUI's checkbox, either by me (with guidance) or another developer with a better knowledge of component structures and customisation?